assets/js/hooks/useChecks.test.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * `useChecks` hook tests.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { renderHook, actHook as act } from '../../../tests/js/test-utils';
+import { muteFetch } from '../../../tests/js/utils';
+import { useChecks } from './useChecks';
+
+describe( 'useChecks', () => {
+	it( 'should return { complete:true, error: undefined } successful check runs.', async () => {
+		muteFetch( /^\/google-site-kit\/v1\/core\/site\/data\/connection/ );
+		const checks = [ async () => Promise.resolve() ];
+		let result;
+		await act( async () => {
+			( { result } = await renderHook( () => useChecks( checks ) ) );
+		} );
+
+		expect( result.current ).toStrictEqual( { complete: true, error: undefined } );
+	} );
+
+	it( 'should return { complete:true, error: "error2" } when first error is encountered.', async () => {
+		muteFetch( /^\/google-site-kit\/v1\/core\/site\/data\/connection/ );
+		const checks = [
+			async () => true,
+			async () => {
+				setTimeout( () => {
+					throw 'error1';
+				}, 1 );
+			},
+			async () => {
+				throw 'error2';
+			},
+		];
+		let result;
+		await act( async () => {
+			( { result } = await renderHook( () => useChecks( checks ) ) );
+		} );
+
+		expect( result.current ).toStrictEqual( { complete: true, error: 'error2' } );
+	} );
+} );
```

Comment:
Let's also add an assertion here that `complete === false` before/while the checks are running.

Also, as mentioned before an empty array of checks should be considered `complete` right away.

Reply:
Updated (some lines above).

Reply:
Updated.